pkgs/by-name/zo/zoho-mail-desktop/package.nix

```
@@ -0,0 +1,55 @@
+{
+  lib,
+  stdenvNoCC,
+  appimageTools,
+  fetchurl,
+}:
+stdenvNoCC.mkDerivation (finalAttrs: {
+  pname = "zoho-mail-desktop";
+  version = "1.7.1";
+
+  src = fetchurl {
+    url = "https://downloads.zohocdn.com/zmail-desktop/linux/zoho-mail-desktop-lite-x64-v${finalAttrs.version}.AppImage";
+    hash = "sha256-KLDJl91vfTdDtUQ5maDuCBU1HJQf4V0VEnplAc4ytZM=";
+  };
+
+  wrapped = appimageTools.wrapType2 { inherit (finalAttrs) pname version src; };
+  extracted = appimageTools.extract { inherit (finalAttrs) pname version src; };
+
+  buildInputs = [
+    finalAttrs.wrapped
+  ];
+
+  dontUnpack = true;
+  dontBuild = true;
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+
+    ln -s ${finalAttrs.wrapped}/bin/zoho-mail-desktop $out/bin/zoho-mail-desktop
+
+    mkdir -p $out/share/applications
+    mkdir -p $out/share/lib/zoho-mail-desktop
+
+    cp -r ${finalAttrs.extracted}/usr/* $out/
+    cp -r ${finalAttrs.extracted}/{locales,resources} $out/share/lib/zoho-mail-desktop/
+
+    cp ${finalAttrs.extracted}/zoho-mail-desktop.desktop $out/share/applications/
+    substituteInPlace $out/share/applications/zoho-mail-desktop.desktop \
+      --replace-fail 'Exec=AppRun' 'Exec=zoho-mail-desktop'
+
+    runHook postInstall
+  '';
```

```suggestion
  extraInstallCommands = ''
    cp ${finalAttrs.extracted}/zoho-mail-desktop.desktop $out/share/applications/
    substituteInPlace $out/share/applications/zoho-mail-desktop.desktop \
      --replace-fail 'Exec=AppRun' 'Exec=zoho-mail-desktop'
  '';
```
this is self explanatory i think. also, if the AppImage has an icon bundled, we can install that too.

---

First of all. Thank you for any suggestions!

zoho needs the bundled icon installed. This was done by 
`cp -r ${extracted}/usr/* $out/`.
Should I make it more explicit?

Also do you think I don't need to copy over the resources and locales?
`cp -r ${finalAttrs.extracted}/{locales,resources} $out/share/lib/zoho-mail-desktop/`

---

From my tests the icons don't get copied over. This is also the workflow mentioned in the [nix manual](https://nixos.org/manual/nixpkgs/stable/#ssec-pkgs-appimageTools-extracting) on that topic.

---

yeah seems like you're right. doing `cp -r ${extracted}/usr/* $out/` is fine then i guess

---

Alright! Will make the changes when I get the time. 

Thank you for all the suggestions, they are very helpful.

---

It does work without 
`cp -r ${finalAttrs.extracted}/{locales,resources} $out/share/lib/zoho-mail-desktop/`
So I'll leave that out.